domain/application/resource/resourcestore_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package resource
+
+import (
+	"context"
+
+	jc "github.com/juju/testing/checkers"
+	"go.uber.org/mock/gomock"
+	gc "gopkg.in/check.v1"
+
+	charmresource "github.com/juju/juju/internal/charm/resource"
+)
+
+type resourceStoreSuite struct {
+	objectStore            *MockObjectStore
+	modelObjectStoreGetter *MockModelObjectStoreGetter
+}
+
+var _ = gc.Suite(&resourceStoreSuite{})
+
+func (s *resourceStoreSuite) setupMocks(c *gc.C) *gomock.Controller {
+	ctrl := gomock.NewController(c)
+
+	s.modelObjectStoreGetter = NewMockModelObjectStoreGetter(ctrl)
+	s.objectStore = NewMockObjectStore(ctrl)
+
+	return ctrl
+}
+
+func (s *resourceStoreSuite) TestObjectStoreGetter(c *gc.C) {
```

Comment:
**suggestion:** Add an error path in testcase of `GetObjectStore` fails and verify that the returned error from GetObjectStore is not hidden (even if we add context)

Reply:
Added.